homeassistant/components/aladdin_connect/cover.py

```diff
@@ -35,30 +42,47 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Aladdin Connect platform."""
+    """Set up Aladdin Connect devices yaml depreciated."""
+    _LOGGER.warning(
+        "Loading aladdin_connect via platform setup is deprecated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-    username: str = config[CONF_USERNAME]
-    password: str = config[CONF_PASSWORD]
-    acc = AladdinConnectClient(username, password)
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Aladdin Connect platform."""
+    username: str = config_entry.data[CONF_USERNAME]
+    password: str = config_entry.data[CONF_PASSWORD]
+    acc = AladdinConnectClient(username, password)
     try:
-        if not acc.login():
+        if not await hass.async_add_executor_job(acc.login):
             raise ValueError("Username or Password is incorrect")
-        add_entities(
-            (AladdinDevice(acc, door) for door in acc.get_doors()),
+        doors = await hass.async_add_executor_job(acc.get_doors)
+        async_add_entities(
+            (AladdinDevice(acc, door) for door in doors),
             update_before_add=True,
         )
     except (TypeError, KeyError, NameError, ValueError) as ex:
         _LOGGER.error("%s", ex)
         persistent_notification.create(
```

Comment:
raise ConfigEntryNotReady instead of a persistent notification and HA will handle it
https://github.com/home-assistant/core/blob/dev/homeassistant/exceptions.py#L114